src/wallet/wallet.cpp

```diff
@@ -4658,3 +4712,78 @@ std::shared_ptr<CWalletTransactionBase> CWalletTransactionBase::MakeWalletObject
         return std::shared_ptr<CWalletTransactionBase>( new CWalletTx(pwallet, dynamic_cast<const CTransaction&>(obj)) );
     }
 }
+
+/* Find unspent notes filtered by payment address, min depth and max depth */
+void CWallet::GetUnspentFilteredNotes(
+    std::vector<CUnspentNotePlaintextEntry>& outEntries,
+    std::set<PaymentAddress>& filterAddresses,
+    int minDepth,
+    int maxDepth,
+    bool requireSpendingKey)
+{
+    LOCK2(cs_main, cs_wallet);
+
+    for (auto & p : mapWallet) {
+        CWalletTransactionBase& wtx = *(p.second);
+
+        // Filter the transactions before checking for notes
+        if (!CheckFinalTx(*wtx.getTxBase()) || !wtx.HasMatureOutputs() || wtx.GetDepthInMainChain() < minDepth || wtx.GetDepthInMainChain() > maxDepth) {
+            continue;
+        }
+
+        if (wtx.mapNoteData.size() == 0) {
+            continue;
+        }
+
+        for (auto & pair : wtx.mapNoteData) {
```

Comment:
Maybe structured binding adds more readibility. Also, imho, derived paired values should be marked const.
```suggestion
        for (const auto & [/*JSOutPoint*/ jsop, /*CNoteData*/ nd] : wtx.mapNoteData) {
```